agent/consul/stream/event.go

```diff
@@ -26,18 +26,14 @@ type Event struct {
 // should not modify the state of the payload if the Event is being submitted to
 // EventPublisher.Publish.
 type Payload interface {
-	// MatchesKey must return true if the Payload should be included in a
-	// subscription requested with the key, namespace, and partition.
-	//
-	// Generally this means that the payload matches the key, namespace, and
-	// partition or the payload is a special framing event that should be
-	// returned to every subscription.
-	MatchesKey(key, namespace, partition string) bool
-
 	// HasReadPermission uses the acl.Authorizer to determine if the items in the
 	// Payload are visible to the request. It returns true if the payload is
 	// authorized for Read, otherwise returns false.
 	HasReadPermission(authz acl.Authorizer) bool
+
+	// TopicKey is used to identify which subscribers should be notified of this
+	// event - e.g. those subscribing to health events for a particular service.
```

Comment:
minor: I think it would be helpful to include what kind of values we generally expect. Ex: for services this is the full service ID, for KV it would be the full key path, etc.  I think that might help readers understand this is simply an opque identifier for the data in the Payload, not some stream-specific tracking ID.

Reply:
Great point! 💡